src/sentry/rules/conditions/level.py

```diff
@@ -58,7 +59,12 @@ def passes(self, event, state, **kwargs):
             return False
 
         desired_level = int(desired_level)
-        level = int(event.group.level)
+        # Fetch the event level from the tags since event.level is
+        # event.group.level which may have changed
+        try:
+            level = LOG_LEVEL_REVERSE_MAP[event.get_tag('level')]
+        except KeyError:
```

Comment:
I don't think we'd need to be cautious here after re-reading code. It seems that we assert the log level is something valid on ingest, and we also apply a default log level in the case of there not being one at all.

In theory, it's possible to send a top level `level` attribute as well as a `level` tag explicitly that doesn't get validated.

This is a secondary problem in that we probably shouldn't add a `level` tag if there already exists a `level` tag. Or disallow duplicate keys entirely.